lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Scholar.Linear.LogisticRegression do
```

Comment:
I'm not sure about the `Linear` subdomain. What does it mean?
The logistic function isn't a linear transformation

Reply:
I don't agree with the statement that logistic regression is not a linear model. https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression and https://stats.stackexchange.com/questions/93569/why-is-logistic-regression-a-linear-classifier.

Reply:
Thanks for the reference. The function is indeed not linear, but since it is reversible they make for Generalized Linear Models: https://scikit-learn.org/stable/modules/linear_model.html#generalized-linear-regression

What I meant by it not being linear is that for `f(x) = logistic(x)`, `f(x + y) != f(x) + f(y)`, which is one of the conditions for linearity (the other one where `f(k*x) = k*f(x)` is also violated)

Reply:
I'm still not totally comfortable with calling this a Linear model, but since it's already called that by SciKit, we can move forward with this namespace :)

Reply:
I totally agree in the sense that this function does not satisfy the conditions of a linear transformation, but as you mentioned it has an inverse function and we can reformulate the problem.